worktrolly-ui/src/app/services/auth.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import firebase from 'firebase/app';
+import { AngularFireAuth } from '@angular/fire/auth';
+import { User } from "../Interface/UserInterface";
+import { Observable } from 'rxjs';
+import { AngularFireFunctions } from '@angular/fire/functions';
+import { map } from 'rxjs/operators';
+import { AngularFirestore, AngularFirestoreDocument } from '@angular/fire/firestore';
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+
+  constructor(public afauth: AngularFireAuth, private functions: AngularFireFunctions, private db: AngularFirestore) { }
+
+  async createUserData(user: User) {
+    const callable = this.functions.httpsCallable('createNewUser');
+    try {
+      const result = await callable({ uid: user.uid, photoURL: user.photoURL, displayName: user.displayName, email: user.email, phoneNumber: user.phoneNumber, providerId: user.providerId }).toPromise();
+      console.log(result);
+    } catch (error) {
+      console.error("Error", error);
+    }
+  }
```

Comment:
Please perform a check into db if the document already exists, and avoid this http request if the document already exists.

Reply:
This needs to be Run Every time when user Logged In or Signed Up bro, because they may have any update in Phone Number ,photoURL ....